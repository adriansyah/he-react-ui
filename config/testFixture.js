// @flow

import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture<T> = {
  component: React.ComponentType<T>,
  name?: string,
};

type FixtureList<T> = Fixture<T>[];

function testOneFixture<T>(fixture: Fixture<T>, variant?: string = 'a') {
  const { component } = fixture;
  const { displayName, name } = component;

  test(`<${displayName || name} /> rendered correctly with ${fixture.name || variant} fixture`, () => {
    const { mount, getWrapper, unmount } = createTestContext({ fixture });

    try {
      mount();
      expect(getWrapper()).toMatchSnapshot();
    } finally {
      unmount();
    }
  });
}

export default function testAllFixtures<T>(fixtures: Fixture<T> | FixtureList<T>, variant?: string) {
  if (!fixtures.forEach) {
    testOneFixture((fixtures: any), variant);
  } else if (typeof fixtures.forEach === 'function') {
    fixtures.forEach(it => testOneFixture((it: any)));
  }
}
