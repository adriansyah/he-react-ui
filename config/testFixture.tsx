import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture<T> = {
  component: React.ComponentType<T>;
  name?: string;
};

type FixtureList<T> = Fixture<T>[];

function testOneFixture<T>(fixture: Fixture<T>, variant: string = 'a') {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;
  mount();
  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => expect(getWrapper()).toMatchSnapshot());
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures<T>(
  fixtures: Fixture<T> | FixtureList<T>,
  variant?: string,
) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as any, variant);
  } else if (typeof fixtures.forEach === 'function') {
    fixtures.forEach(it => testOneFixture(it as any));
  }
}
