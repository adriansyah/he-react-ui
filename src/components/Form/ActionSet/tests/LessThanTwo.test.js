// @flow
import testFixture from '../../../../../config/testFixture';
import fixture from '../fixtures/LessThanTwo.fixture.js';

testFixture(fixture, 'LessThanTwo');
