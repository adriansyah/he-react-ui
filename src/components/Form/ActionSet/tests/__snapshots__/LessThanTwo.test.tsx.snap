// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnClickOutside(ActionSet) /> rendered correctly with LessThanTwo fixture 1`] = `
<OnClickOutside(ActionSet)
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
>
  <ActionSet
    className=""
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    expanded={false}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
  >
    <div
      className="outer"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="actions"
      >
        <div
          className="action"
          key="0/.0"
          onClick={[Function]}
        >
          <ActionIcon
            className=""
            color="red"
            disabled={false}
            icon="Delete"
            onClick={[Function]}
            title="Delete"
          />
        </div>
        <div
          className="action"
          key="1/.1"
          onClick={[Function]}
        >
          <ActionIcon
            className=""
            color="teal"
            disabled={false}
            icon="Edit"
            onClick={[Function]}
            title="Edit"
          />
        </div>
      </div>
    </div>
  </ActionSet>
</OnClickOutside(ActionSet)>
`;
